app/src/main/java/app/wefridge/wefridge/EditFragment.kt

```diff
@@ -17,39 +51,548 @@ import app.wefridge.wefridge.placeholder.PlaceholderContent
+                            setAddressStringToItemAddressTextEdit(GeoPoint(location.latitude, location.longitude))
+                            itemAddressTextInputLayout.editText?.clearFocus()
+                        } else {
+                            displayAlertDialogOnFailedLocationDetermination()
+                        }
+                    }
+
+                    .addOnFailureListener {
+                        displayAlertDialogOnPermissionDenied()
+                    }
+
+            }
+            shouldShowRequestPermissionRationale(Manifest.permission.ACCESS_FINE_LOCATION) -> {
+                // called when permission denied
+                displayAlertDialogOnFailedLocationDetermination()
+            }
+            else -> {
+                // called when permission settings unspecified (like "ask every time")
+                requestPermissionLauncher.launch(
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                )
+            }
+        }
+
+    }
+
+    private fun setAddressStringToItemAddressTextEdit(location: GeoPoint) {
+        // the following code is based on https://stackoverflow.com/questions/9409195/how-to-get-complete-address-from-latitude-and-longitude
+
+        itemAddressTextInputLayout.editText?.setText(buildAddressString(location))
+    }
+
+    private fun getGeoPointFromAddressUserInput(): GeoPoint? {
+
+        // the following piece of code is inspired by https://stackoverflow.com/questions/3574644/how-can-i-find-the-latitude-and-longitude-from-address/27834110#27834110
+        val userInputAddress = itemAddressTextInputLayout.editText?.text.toString()
+        val geocoder = Geocoder(requireContext())
+
+        if (userInputAddress == "") return null
+        val matchedAddresses: List<Address> = geocoder.getFromLocationName(userInputAddress, 1)
+        if (matchedAddresses.isEmpty()) return null
+
+
+        val chosenAddress = matchedAddresses[0]
+
+        return GeoPoint(chosenAddress.latitude, chosenAddress.longitude)
+
+    }
+
+    private fun buildAddressString(location: GeoPoint): String {
+        val geocoder = Geocoder(requireContext(), Locale.getDefault())
+        val address =
+            geocoder.getFromLocation(location.latitude, location.longitude, 1)
+        val addressLine = address.get(0).getAddressLine(0)
+        val city = address.get(0).locality
+        val state = address.get(0).adminArea
+        val postalCode = address.get(0).postalCode
+
+        return "${addressLine}, ${postalCode} ${city}, ${state}"
+    }
+
+    private fun displayAlertOnSaveSharedItemWithoutContactEmail() {
+        AlertDialog.Builder(requireContext())
+            .setTitle("Contact email missing")
+            .setMessage("In order to share an Item, you have to provide a contact email other users can use to reach out to you. Please state your email address in the settings tab.")
+            .setPositiveButton(android.R.string.ok, null)
+            .show()
```

Comment:
Create issue to remove strings in this file

Reply:
Do you mind if I put this "todo" into the refactoring issue or would you like to have a seperate issue for that?

Reply:
Separate issue was my intention

Reply:
Okay, I created a new issue for that: #43